src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -252,6 +406,46 @@ def get_task_res(self, task_ids: set[UUID]) -> list[TaskRes]:
 
         return list(ret.values())
 
+    def get_message_res(self, message_ids: set[UUID]) -> list[Message]:
+        """Get reply Messages for the given Message IDs."""
+        ret: dict[UUID, Message] = {}
+
+        with self.lock:
+            current = time.time()
+
+            # Verify Messge IDs
+            ret = verify_message_ids(
+                inquired_message_ids=message_ids,
+                found_message_ins_dict=self.message_ins_store,
+                current_time=current,
+            )
+
+            # Find all reply Messages
+            message_res_found: list[Message] = []
+            for message_id in message_ids:
+                # If Message exists and is not delivered, add it to the list
+                if message_res_id := self.message_ins_id_to_message_res_id.get(
+                    message_id
+                ):
+                    message_res = self.message_res_store[message_res_id]
+                    if message_res.metadata.delivered_at == "":
+                        message_res_found.append(message_res)
+            tmp_ret_dict = verify_found_message_replies(
+                inquired_message_ids=message_ids,
+                found_message_ins_dict=self.message_ins_store,
+                found_message_res_list=message_res_found,
+                current_time=current,
+            )
+            ret.update(tmp_ret_dict)
+
+            # Mark existing reply Messages to be returned as delivered
+            delivered_at = now().isoformat()
+            for message_res in message_res_found:
+                # pylint: disable-next=W0212
+                message_res.metadata._delivered_at = delivered_at  # type: ignore
```

Comment:
Why does this differ from the above?

```suggestion
                message_res.metadata.delivered_at = delivered_at
```

Reply:
right. I'll update it. at first i didn't plan to add setters for `delivered_at` but i did include them in the end.